### PR TITLE
Refactor Scripts

### LINKED ISSUE #138: Refactor Scripts
- Replace all the shell scripts with python scripts.
    - Docker controll -> Docker Python SDK
    - MongoDB -> PyMongo
- Remove all the DAO/DTO related code (we don't need to use DTO pattern with an already highly encapsulated DB library (mongo), it doesn't make sense).

### PR DESCRIPTION
**Pull request by YuanmingLeee**
Closes #138 

- [x] Start MongoDB service
- [x] Start cAdvisor service
- [x] Start Node exporter service
- [x] Start all services
- [x] Cleanup
- [x] Update doc

#### Before submitting this PR, please make sure you have:

- [x] run `python -m pytest tests/` and confirmed all checks still pass.
- [x] verified that any code or assets from external sources are properly credited.

### CODE REVIEW

File: modelci/client.py [hunk 1]
```diff
@@ -0,0 +1,224 @@
+#!/usr/bin/env python
```
Comment (huangyz0918) [inline modelci/client.py:1]: should not be named as client, since we already have many clients for model profiling.

**Review by huangyz0918 — CHANGES_REQUESTED** [review]

File: modelci/client.py (hunk 1 above)
Comment (YuanmingLeee) [reply modelci/client.py:1]: does refactor client to runs looks fine?
From mlflow https://github.com/mlflow/mlflow/blob/master/mlflow/runs.py

File: modelci/runs.py [hunk 2]
```diff
@@ -0,0 +1,308 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
```
Comment (huangyz0918) [inline modelci/runs.py:2]: I recommend using something like `ContainerManager` as a class here, and merge the `docker_api_utils.py` with this file.

File: modelci/init-mongo.sh [hunk 3]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env sh
```
Comment (huangyz0918) [inline modelci/init-mongo.sh:1]: can this replaced by pymongo commands?

File: README.md [hunk 4]
```diff
@@ -143,7 +153,7 @@ MLModelCI provides a dispatcher to deploy a model as a cloud service. The dispat
 We search for a converted model and then dispatch it to a device with a specific batch size.
 
 ```python
-from modelci.hub.deployer.serving import serve
```
Comment (huangyz0918) [inline README.md:146]: updating documents are required.

File: modelci/runs.py [hunk 5]
```diff
@@ -0,0 +1,308 @@
...
+        try:
+            os.killpg(os.getpgid(pid), signal.ITIMER_REAL)
+            logger.info(f'Already running, found PID={pid}.')
+            return pid
+        except OSError:
+            pass
+    return False
+
+
+def _stop_service(docker_client, label):
```
Comment (huangyz0918) [inline modelci/runs.py:95]: what does '-' prefix mean in function name? Is this function private to other classes or files? if not, please remove them.

File: modelci/runs.py [hunk 6]
```diff
@@ -0,0 +1,308 @@
...
+            else:
+                args1 = ('locate', 'libnvidia-ml.so.1')
+                args2 = ('grep', '-v', 'lib32')
+                args3 = ('head', '-1')
+                locate = subprocess.Popen(args1, stdout=subprocess.PIPE)  # NOQA: B603
+                grep = subprocess.Popen(args2, stdin=locate.stdout, stdout=subprocess.PIPE)
+                locate.wait()
+                grep.wait()
+                # NOQA: B603
+                lib_path = subprocess.check_output(args3, stdin=grep.stdout, universal_newlines=True).strip()
```
Comment (huangyz0918) [inline modelci/runs.py:185]: change `lib_path` to `libnvidia_ml_path`.

File: modelci/init-mongo.sh (hunk 3 above)
Comment (YuanmingLeee) [reply modelci/init-mongo.sh:1]: A more proper way is using a init bash script or a init JS file.
Providing a init file to the container is recommended, see https://stackoverflow.com/a/54064268/13173608

File: modelci/runs.py (hunk 5 above)
Comment (YuanmingLeee) [reply modelci/runs.py:95]: A private method

Comment (huangyz0918) [reply modelci/runs.py:95]: you should build a class so we can call that private. According to my comment above.

File: modelci/init-mongo.sh (hunk 3 above)
Comment (huangyz0918) [reply modelci/init-mongo.sh:1]: Let's use pymongo, it's much better that his solution since the question did indicate the programming language.

Comment (huangyz0918) [reply modelci/init-mongo.sh:1]: And their case is for docker container, for us we don't care mongo service is a local service or container service, a python side check will be better.

Comment (YuanmingLeee) [reply modelci/init-mongo.sh:1]: Added. Meanwhile, is it good to keep `init-db.sh`, as this init file can be used in `docker-compose` later?

File: modelci/runs.py (hunk 6 above)
Comment (YuanmingLeee) [reply modelci/runs.py:185]: refactored

**Review by huangyz0918 — APPROVED** [review]

**Pull request merged**